File: src/main/java/com/bot/model/enumeration/stats/CommonStatisticType.java
package com.bot.model.enumeration.stats;

import com.bot.model.enumeration.DataAction;
import lombok.Getter;

import java.util.Arrays;

@Getter
public enum CommonStatisticType implements Type {
    CATEGORY("По категориям", "cat"),
    DYNAMICS("Динамика", "dyn");

    CommonStatisticType(String ruName, String name) {
        this.ruName = ruName;
        this.name = name;
    }

    private String ruName;
    private String name;

    public DataAction getByRuName(String ruName){
        return Arrays.stream(DataAction.values())
                .filter(dataAction -> dataAction.getRuName().toLowerCase().equals(ruName.toLowerCase()))
                .findFirst()
                .orElse(null);
    }

    public DataAction getByName(String name){
        return Arrays.stream(DataAction.values())
                .filter(dataAction -> dataAction.getRuName().toLowerCase().equals(name.toLowerCase()))
                .findFirst()
                .orElse(null);
    }
}

File: telegram-adapter/src/main/kotlin/com/planner/model/enumeration/stats/CommonStatisticType.kt
package com.planner.model.enumeration.stats

import com.planner.model.enumeration.DataAction
import java.util.*

enum class CommonStatisticType(var ruName: String, val label: String){
    CATEGORY("По категориям", "cat"),
    DYNAMICS("Динамика", "dyn");

    fun getByRuName(ruName: String): DataAction? {
        return Arrays.stream(DataAction.values())
            .filter { dataAction: DataAction? ->
                dataAction!!.ruName.lowercase(Locale.getDefault()) == ruName.lowercase(
                    Locale.getDefault()
                )
            }
            .findFirst()
            .orElse(null)
    }

    fun getByName(name: String): DataAction? {
        return Arrays.stream(DataAction.values())
            .filter { dataAction: DataAction? ->
                dataAction!!.ruName.lowercase(Locale.getDefault()) == name.lowercase(
                    Locale.getDefault()
                )
            }
            .findFirst()
            .orElse(null)
    }
}